framework/src/fwk_module.c

```diff
@@ -449,6 +450,86 @@ int fwk_module_start(void)
     return FWK_SUCCESS;
 }
 
+static int fwk_module_stop_elements(struct fwk_module_context *module_ctx)
+{
+    int status;
+    const struct fwk_module *module;
+    unsigned int element_idx;
+
+    module = module_ctx->desc;
+
+    for (element_idx = 0; element_idx < module_ctx->element_count;
+         element_idx++) {
+
+        if (module->stop != NULL) {
+            status = module->stop(
+                        fwk_id_build_element_id(module_ctx->id, element_idx));
+            if (!fwk_expect(status == FWK_SUCCESS)) {
+                FWK_LOG_CRIT(fwk_module_err_msg_func, status, __func__);
+                return status;
+            }
+        }
+
+        module_ctx->element_ctx_table[element_idx].state =
+            FWK_MODULE_STATE_SUSPENDED;
+    }
+
+    return FWK_SUCCESS;
+}
+
+static int fwk_module_stop_module(struct fwk_module_context *module_ctx)
+{
+    int status;
+    const struct fwk_module *module;
+
+    module = module_ctx->desc;
+
+    if (module->stop != NULL) {
+        status = module->stop(module_ctx->id);
+        if (!fwk_expect(status == FWK_SUCCESS)) {
+            FWK_LOG_CRIT(fwk_module_err_msg_func, status, __func__);
+            return status;
+        }
+    }
+
+    module_ctx->state = FWK_MODULE_STATE_SUSPENDED;
+
+    return fwk_module_stop_elements(module_ctx);
```

Comment:
Do not call this function if `module->stop == NULL`

Reply:
as said above, I have followed fwk_module_start_module skeleton